packages/amplication-git-utils/src/git/git.service.ts

```diff
@@ -89,39 +99,213 @@ export class GitClientService {
     }
 
     if (pullRequestMode === EnumPullRequestMode.Accumulative) {
-      await this.provider.createBranchIfNotExists({
+      const gitClient = new GitClient();
+      const cloneFolder = process.env.CLONES_FOLDER;
+      if (!cloneFolder) {
+        throw new MissingEnvParam("CLONES_FOLDER");
+      }
+
+      const randomUUID = v4();
+
+      const cloneUrl = this.provider.getCloneUrl({
+        owner,
+        repositoryName,
+      });
+
+      const cloneDir = normalize(
+        join(cloneFolder, this.provider.name, owner, repositoryName, randomUUID)
+      );
+
+      await gitClient.clone(cloneUrl, cloneDir);
+
+      await this.restoreAmplicationBranchIfNotExists({
         owner,
         repositoryName,
         branchName,
-      }),
-        await this.provider.createCommit({
+        gitClient,
+      });
+
+      const diffFolder = normalize(
+        join(
+          `.amplication/diffs`,
+          this.provider.name,
           owner,
+
           repositoryName,
-          commitMessage,
-          branchName,
-          files: preparedFiles,
+          randomUUID
+        )
+      );
+
+      const { diff } = await this.preCommitProcess({
+        branchName,
+        gitClient,
+        owner,
+        repositoryName,
+      });
+      let fullDiffPath = "";
+      if (diff) {
```

Comment:
you can take any if and put them in a function and pass it the diff. it will be cleaner and without diff. another option is composing all this actions in function. it will give you the option to add new actions in the lineup process and much cleaner and decoupled